idunn/utils/logging.py

```diff
@@ -0,0 +1,38 @@
+from .settings import Settings
+from pythonjsonlogger import jsonlogger
+import logging
+import json
+from apistar import http
+
+def init_logging(settings: Settings):
+    """
+    init the logging for the server
+    """
+    log_format = settings['LOG_FORMAT']
+    as_json = settings['LOG_JSON']
+
+    levels = settings['LOG_LEVEL_BY_MODULE']
+    for module, lvl in json.loads(levels).items():
+        log_level = lvl.upper()
+        log_level = logging.getLevelName(log_level)
+        
+        logger = logging.getLogger(module)
+        logger.setLevel(log_level)
+
+    logHandler = logging.StreamHandler()
+    if as_json:
+        formatter = jsonlogger.JsonFormatter(log_format)
+        logHandler.setFormatter(formatter)
+    else:
+        logHandler.setFormatter(logging.Formatter(log_format))
+    
+    # we set this handler to the main logger
+    logging.getLogger().handlers = [logHandler]
+
+class LogErrorHook:
+    def on_response(self, response: http.Response, exc: Exception):
+        if exc is not None:
+            logging.getLogger('idunn.error').exception("handler returned an error")
```

Comment:
Here `exc` is necessarily a subclass of `HTTPException`, that is directly handled by apistar to build a response.
As a result any 404 (for example) is logged here as an ERROR, which may be not be relevant.

Should we keep the `on_error` hook only ?